src/fs.cpp

```diff
@@ -147,11 +147,11 @@ std::string get_filesystem_error_message(const fs::filesystem_error& e)
 #ifdef WIN32
 #ifdef __GLIBCXX__
 
-// reference: https://github.com/gcc-mirror/gcc/blob/gcc-7_3_0-release/libstdc%2B%2B-v3/include/std/fstream#L270
+// reference: https://github.com/gcc-mirror/gcc/blob/releases%2Fgcc-7.3.0/libstdc%2B%2B-v3/include/std/fstream#L270
 
 static std::string openmodeToStr(std::ios_base::openmode mode)
 {
-    switch (mode & ~std::ios_base::ate) {
+    switch (static_cast<int>(mode & ~std::ios_base::ate)) { // casting is required to suppress -Wswitch warnings
```

Comment:
`std::ios_base::ate` and friends are defined as `unsigned int` on my platform. What is the warning that this tries to fix?

Reply:
> `std::ios_base::ate` and friends are defined as `unsigned int` on my platform.

https://github.com/gcc-mirror/gcc/blob/releases/gcc-7.3.0/libstdc%2B%2B-v3/include/bits/ios_base.h#L111-L122

> What is the warning that this tries to fix?

https://gcc.gnu.org/onlinedocs/gcc-9.3.0/gcc/Warning-Options.html#Warning-Options:
> `-Wswitch`
> ... `case` labels outside the enumeration range also provoke warnings when this option is used (even if there is a `default` label).

Reply:
Ok, we are fine because if some of the case values is out of the range of `int`, then we would get a warning:
```cpp
void f(unsigned int x)
{
    const unsigned int y = 3000000000U;

    switch (static_cast<int>(x)) { 
    case y:
        std::cout << "foo\n";
        break;
    } 
}
```

```
error: case value evaluates to 3000000000, which cannot be narrowed to type 'int' [-Wc++11-narrowing]
```

from clang 12 with `-Wall`